cristin-commons/src/main/java/no/unit/nva/cristin/common/client/ApiClient.java

```diff
@@ -135,19 +152,30 @@ public HttpResponse<String> fetchQueryResults(URI uri) throws ApiGatewayExceptio
     }
 
     protected HttpResponse<String> getSuccessfulResponseOrThrowException(HttpRequest httpRequest)
-        throws GatewayTimeoutException, FailedHttpRequestException {
+        throws FailedHttpRequestException {
 
         try {
-            return client.send(httpRequest, BodyHandlers.ofString(StandardCharsets.UTF_8));
-        } catch (HttpTimeoutException timeoutException) {
-            logError(ERROR_MESSAGE_BACKEND_FAILED_WITH_EXCEPTION, httpRequest.uri().toString(), timeoutException);
-            throw new GatewayTimeoutException();
-        } catch (IOException | InterruptedException otherException) {
-            logError(ERROR_MESSAGE_BACKEND_FAILED_WITH_EXCEPTION, httpRequest.uri().toString(), otherException);
-            throw new FailedHttpRequestException(ErrorMessages.ERROR_MESSAGE_BACKEND_FETCH_FAILED);
+            return fetchResponseWithRetry(httpRequest);

```

Comment:
the calling method `getSuccessfulResponseOrThrowException `is shared between GET, POST and PATCH. Should apply to most of the http requests across the whole proxy as ApiClient is extended everywhere